File: src/main/java/com/dh/ondot/schedule/infra/dto/KakaoSearchRoadAddressResponse.java
package com.dh.ondot.schedule.infra.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public record KakaoSearchRoadAddressResponse(
        @JsonProperty("documents") List<DocumentsResponse> documents
) {
    public record DocumentsResponse(
            @JsonProperty("road_address") RoadAddressResponse roadAddress
    ) {
        public record RoadAddressResponse(
                @JsonProperty("address_name") String addressName,
                @JsonProperty("building_name") String buildingName,
                @JsonProperty("x") String x,
                @JsonProperty("y") String y
        ) {
        }
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/dto/KakaoSearchRoadAddressResponse.kt
package com.dh.ondot.schedule.infra.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class KakaoSearchRoadAddressResponse(
    @param:JsonProperty("documents") val documents: List<DocumentsResponse>,
) {
    data class DocumentsResponse(
        @param:JsonProperty("road_address") val roadAddress: RoadAddressResponse?,
    ) {
        data class RoadAddressResponse(
            @param:JsonProperty("address_name") val addressName: String,
            @param:JsonProperty("building_name") val buildingName: String?,
            @param:JsonProperty("x") val x: String?,
            @param:JsonProperty("y") val y: String?,
        )
    }
}
